tests/integration_test.go

```diff
@@ -0,0 +1,680 @@
+			{
+				Name: aws.String("tag:Name"),
+				Values: []*string{
+					aws.String(awsTag),
+				},
+			},
+		},
+	}
+
+	ec2Result, err := ec2Client.DescribeInstances(ec2DescInp)
+	if err != nil {
+		t.Fatal("There was an error. ", err)
+	}
+
+	// then
+	if len(ec2Result.Reservations[0].Instances) != 1 {
+		t.Error("Expected ", instancesNumber, "instance, got ", len(ec2Result.Reservations[0].Instances))
+	}
+
+}
+
+func TestOnDestroyPlanShouldDisplayDestroyPlan(t *testing.T) {
+	// given
+	expectedOutputRegexp := "Plan: 0 to add, 0 to change, 14 to destroy"
+
+	// when
+	stdout, stderr := runDocker(t, "plan-destroy", awsAccessKey, awsSecretKey)
+
+	if stderr.Len() > 0 {
+		t.Fatal("There was an error during executing a command. ", string(stderr.Bytes()))
+	}
+
+	outStr := string(stdout.Bytes())
+
+	matched, err := regexp.MatchString(expectedOutputRegexp, outStr)
+	if err != nil {
+		t.Fatal("There was an error matching expression: ", err)
+	}
+
+	// then
+	if !matched {
+		t.Error("Expected to find expression matching:\n", expectedOutputRegexp, "\nbut found:\n", outStr)
+	}
+}
+
+func TestOnDestroyShouldDestroyEnvironment(t *testing.T) {
+	// given
+	expectedOutputRegexp := "Apply complete! Resources: 0 added, 0 changed, 14 destroyed."
+
+	// when
+	stdout, stderr := runDocker(t, "destroy", awsAccessKey, awsSecretKey)
+
+	if stderr.Len() > 0 {
+		t.Fatal("There was an error during executing a command. ", string(stderr.Bytes()))
+	}
+
+	outStr := string(stdout.Bytes())
+
+	matched, err := regexp.MatchString(expectedOutputRegexp, outStr)
+	if err != nil {
+		t.Fatal("There was an error matching expression: ", err)
+	}
+
+	// then
+	if !matched {
+		t.Error("Expected to find expression matching:\n", expectedOutputRegexp, "\nbut found:\n", outStr)
+	}
+}
+
+func setup() {
+	log.Println("Initialize test")
+	awsAccessKey = os.Getenv("AWS_ACCESS_KEY_ID")
+	if len(awsAccessKey) == 0 {
+		log.Fatalf("expected non-empty AWS_ACCESS_KEY_ID environment variable")
+	}
+	awsAccessKey = "M_AWS_ACCESS_KEY=" + awsAccessKey
+
+	awsSecretKey = os.Getenv("AWS_SECRET_ACCESS_KEY")
+	if len(awsSecretKey) == 0 {
+		log.Fatalf("expected non-empty AWS_SECRET_ACCESS_KEY environment variable")
+	}
+	awsSecretKey = "M_AWS_SECRET_KEY=" + awsSecretKey
+
+	imageTag = os.Getenv("AWSBI_IMAGE_TAG")
+	if len(imageTag) == 0 {
+		log.Fatalf("expected non-empty AWSBI_IMAGE_TAG environment variable")
+	}
+
+	err := os.MkdirAll(sharedAbsoluteFilePath, os.ModePerm)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Generating Keys")
+	err = generateRsaKeyPair(sharedAbsoluteFilePath, sshKeyName)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func cleanupDiskTestStructure() {
+	log.Println("Starting cleanup.")
+	err := os.RemoveAll(sharedAbsoluteFilePath)
+	if err != nil {
+		log.Fatal("Cannot remove data folder. ", err)
+	}
+	log.Println("Cleanup finished.")
+}
+
+func cleanupAWSResources() {
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(awsRegion)})
```

Comment:
```suggestion
	newSession, err := session.NewSession(&aws.Config{Region: aws.String(awsRegion)})
```
collides with imported package name